server/events.go

```diff
@@ -66,28 +66,51 @@ type DataStats struct {
 	Bytes int64 `json:"bytes"`
 }
 
-// This will send a message.
-// TODO(dlc) - Note we want the sequence numbers to be serialized but right now they may not be.
-func (s *Server) sendInternalMsg(r *SublistResult, subj string, msg []byte) {
+// Used for internally queueing up messages that the server wants to send.
+type pubMsg struct {
+	r   *SublistResult
+	sub string
+	msg []byte
+}
+
+func (s *Server) internalSendLoop() {
+	defer s.grWG.Done()
+	s.mu.Lock()
 	if s.sys == nil {
 		return
```

Comment:
Unlock missing here.

Reply:
Yea already saw that. Also going to make sure sequence numbers are sequential for outgoing messages so am looking at that code atm.